src/context/ServiceInfoContext.tsx

```diff
@@ -74,15 +79,19 @@ const ServiceInfoProvider = ({ children }: React.PropsWithChildren<{}>) => {
 
             const shouldResetState = currentWallet && (!activeWalletName || currentWallet.name !== activeWalletName)
             if (shouldResetState) {
-              setCurrentWallet(null)
-              clearSession()
+              resetWalletAndClearSession()
             }
           }
           return info
         })
         .catch((err) => {
           if (!signal.aborted) {
-            setConnectionError(err)
+            const shouldResetState = err instanceof Api.JmApiError && err.response.status === 401
+            if (shouldResetState) {
```

Comment:
Nit I'd love to hear your opinion on:

I'd personally either inline the `err instanceof ...` check or, preferably, name the variable `isUnauthorized`.

My thinking is that it's more clear to have the if-statement read:

"If the request is unauthorized, reset the wallet and clear the session."

Compared to:

"If we should reset the state, reset the wallet and clear the session."

Since the former contains a more clear picture of what's happening compared to the latter. With the variable named `shouldResetState` we're basically moving the if-statement "logic" to the variable assignment and the if-statement itself is not convening much information anymore (at least when it comes to naming).

This is a super esoteric discussion of course and _not_ a change request. I'd still love to hear your opinion on it. I have a soft spot for obsucre naming discussions sometimes. 😄

Reply:
Affirmative. Thanks for taking the time to lay down your thoughts. Totally reasonable and improves readability a lot. 
My opinion is, that I should be more thoughtful about initial variable naming. I sometimes think I can revisit it in the future and end up never doing it. Improving these habits is a skill worth practicing. You'll notice in the future if I could live up to it. Please always mention if you see an opportunity to increase readability.